nixos/modules/services/networking/corerad.nix

```diff
@@ -4,18 +4,58 @@ with lib;
 
 let
   cfg = config.services.corerad;
+
+  writeTOML = name: x:
+    pkgs.runCommandNoCCLocal name {} ''
+      echo '${builtins.toJSON x}' | ${pkgs.go-toml}/bin/jsontoml > $out
+    '';
+
+  # Prefer settings attributes over a config file if both are set.
+  coreradTOML = if cfg.settings != null then
+    writeTOML "corerad.toml" cfg.settings
+  else
+    cfg.configFile;
+
 in {
-  meta = {
-    maintainers = with maintainers; [ mdlayher ];
-  };
+  meta.maintainers = with maintainers; [ mdlayher ];
 
   options.services.corerad = {
     enable = mkEnableOption "CoreRAD IPv6 NDP RA daemon";
 
+    settings = mkOption {
+      default = null;
+      type = types.attrs;
+      example = literalExample ''
+        {
+          interfaces = [
+            # eth0 is an upstream interface monitoring for IPv6 router advertisements.
+            {
+              name = "eth0";
+              monitor = true;
+            }
+            # eth1 is a downstream interface advertising IPv6 prefixes for SLAAC.
+            {
+              name = "eth1";
+              advertise = true;
+              prefix = [{ prefix = "::/64"; }];
+            }
+          ];
+          # Optionally enable Prometheus metrics.
+          debug = {
+            address = "localhost:9430";
+            prometheus = true;
+          };
+        }
+      '';
+      description = ''
+        Configuration for CoreRAD, see <link xlink:href="https://github.com/mdlayher/corerad/blob/master/internal/config/default.toml"/> for supported values.'';
+    };
+
     configFile = mkOption {
       type = types.path;
       example = literalExample "\"\${pkgs.corerad}/etc/corerad/corerad.toml\"";
-      description = "Path to CoreRAD TOML configuration file.";
+      description =
+        "Path to CoreRAD TOML configuration file. Ignored if settings is in use.";
```

Comment:
This should be the other way around: If `configFile` is set, `settings` should be ignored. This allows people to override whatever is auto-generated if desired. The best way to do this is by assigning
```nix
{
  services.corerad.configFile = mkDefault (writeTOML "corerad.toml" cfg.settings);
}
```